src/passes/Precompute.cpp

```diff
@@ -203,26 +206,24 @@ struct Precompute
         // this expression causes a return. if it's already a return, reuse the
         // node
         if (auto* ret = curr->dynCast<Return>()) {
-          if (flow.getSingleValue().type != Type::none) {
+          if (flow.values.size() > 0) {
             // reuse a const value if there is one
-            if (ret->value) {
+            if (ret->value && flow.values.size() == 1) {
               if (auto* value = ret->value->dynCast<Const>()) {
                 value->value = flow.getSingleValue();
                 value->finalize();
                 return;
               }
             }
-            ret->value =
-              Builder(*getModule()).makeConstExpression(flow.getSingleValue());
+            ret->value = flow.getConstExpression(*getModule());
```

Comment:
can we remove the `Const`-handling code just above here? I think `getConstExpression` will make either a `Const` or a `MakeTuple` for us already.

Reply:
Yes, although we could have done so before this change as well. My understanding is that the special Const handing is just to save an allocation of a new IR Const node. Do you want to remove that memory optimization?

Reply:
Right, thanks, it's for that reason. Let's leave it alone then.